File: 520-DetectCapital/520-DetectCapital.py
# Last updated: 30/08/2026, 20:01:15
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        count = 0
        for i in word:
            if i.isupper():
                count+=1

        if count == 0 or count == len(word):
            return True
        elif count == 1 and word[0].isupper():
            return True
        return False

File: 520-DetectCapital/520-DetectCapital.cpp
// Last updated: 30/08/2026, 20:01:11
class Solution {
public:
    bool detectCapitalUse(string word) {
        int count = 0;
        for(int i = 0; i< word.size(); i++){
            if (isupper(word[i])){
                count++;
            }
        }
        if ((count == word.size()) || count ==0){
            return true;
        }
        else if (count == 1){
            if (isupper(word[0])){
                return true;
            }
        }
        return false;
        
    }
};